src/analyzer/protocol/http/HTTP.cc

```diff
@@ -1489,6 +1485,13 @@ const String* HTTP_Analyzer::UnansweredRequestMethod()
 
 int HTTP_Analyzer::HTTP_ReplyLine(const char* line, const char* end_of_line)
 	{
+	// There isn't any validation for reply data from an HTTP/0.9 request.
+	if ( request_version.major == 0 && request_version.minor == 9 )
```

Comment:
I think this can work as a fix. I do wonder now a bit about the following HTTP message consumption because
1) the first line of the actual content was consumed as a reply line 
2) it looks like we'd parse the rest of the response as a MIME message which isn't true for HTTP/0.9

So, I'm almost tempted to suggest turning off the HTTP analyzer in addition to the added early return. Possibly also moving the 0.9 version check below into the `PrefixMatch` condition?

I think as a follow-up it may be worthwhile to discuss if HTTP/0.9 should be a separate analyzer altogether to remove the intertwining :-(

Reply:
> it looks like we'd parse the rest of the response as a MIME message which isn't true for HTTP/0.9

Yeah, this is definitely not good. Maybe just implement a deliver method for HTTP_Message that sidesteps a bunch of the processing if it's 0.9?

> So, I'm almost tempted to suggest turning off the HTTP analyzer in addition to the added early return. Possibly also moving the 0.9 version check below into the PrefixMatch condition?

I assume you mean in `HTTP_ReplyLine`? Falling into that if statement was what got us here. I think a fast exit just setting the version is the cleanest way out.

Reply:
> Maybe just implement a deliver method for HTTP_Message that sidesteps a bunch of the processing if it's 0.9?

I could see that as a follow-up, too, depending on how hairy it gets?

> I assume you mean in HTTP_ReplyLine? Falling into that if statement was what got us here. I think a fast exit just setting the version is the cleanest way out.

Right, I thought within the "does not start with "HTTP/" check to do the HTTP 0.9 request test and allow success. Mostly as a heuristic if there's a 0.9 request with a 1.0/1.1 response, but given we have not seen that, probably better to skip.

Reply:
> I could see that as a follow-up, too, depending on how hairy it gets?

Added in the latest commit.